packages/@aws-cdk/aws-appmesh/lib/virtual-gateway-listener.ts

```diff
@@ -220,8 +179,36 @@ class GrpcGatewayListener extends VirtualGatewayListener {
           port: this.port,
           protocol: Protocol.GRPC,
         },
-        healthCheck: this.healthCheck? this.renderHealthCheck(this.healthCheck): undefined,
+        healthCheck: this.healthCheck ? renderHealthCheck(this.healthCheck, this.protocol, this.port): undefined,
       },
     };
   }
-}
\ No newline at end of file
+}
+
+function renderHealthCheck(
+  hc: HealthCheck, listenerProtocol: Protocol, listenerPort: number): CfnVirtualGateway.VirtualGatewayHealthCheckPolicyProperty {
+
+  if (hc.protocol === Protocol.TCP) {
```

Comment:
I wonder - do we need these checks yet? I think right now it should not be possible to construct a `TCP` Listener with a health check, right?

Same for the checks below?

Reply:
I think we do right now since the `HealthCheck` interface does not contain protocol variants. That means you can pass in a TCP health check for a VirtualGateway even though it is not correct.

```
      new appmesh.VirtualGateway(stack, 'httpGateway', {
        mesh: mesh,
        listeners: [appmesh.VirtualGatewayListener.http({
          port: 443,
          healthCheck: {
            protocol: appmesh.Protocol.TCP,
            interval: cdk.Duration.seconds(10),
          },
        })],
      });
```

Would throw an exception: `An error occurred (BadRequestException) when calling the CreateVirtualGateway operation: HealthCheck protocol must be one of [http,http2,grpc].`

Reply:
I need to create an issue so we can have the health checks migrated to the same static method strategy we use elsewhere 😄